Source/WebCore/platform/graphics/MediaPlayer.cpp

```diff
@@ -1229,7 +1229,7 @@ void MediaPlayer::networkStateChanged()
     // let the next engine try.
     if (m_private->networkState() >= MediaPlayer::NetworkState::FormatError && m_private->readyState() < MediaPlayer::ReadyState::HaveMetadata) {
         client().mediaPlayerEngineFailedToLoad();
-        if (!m_activeEngineIdentifier && installedMediaEngines().size() > 1 && (m_contentType.isEmpty() || nextBestMediaEngine(m_currentMediaEngine))) {
+        if (!m_activeEngineIdentifier && installedMediaEngines().size() > 1 && !m_contentType.isEmpty() && nextBestMediaEngine(m_currentMediaEngine)) {
```

Comment:
The original behaviour of loading with next media engine 
 in reloadTimerFired() when contentType is empty has been there since https://bugs.webkit.org/show_bug.cgi?id=58462, more than 10 years ago.
 Do you remember why this change was needed?

Reply:
This is needed to notify html media element about the error, and preserve the player with an actual error message. otherwise, it just silently consumes the error while trying to load next engine.

Reply:
This caused regressions in Apple ports upstream, so I reverted the line to avoid player private regeneration and added code to cache the player private error in MediaPlayer as m_lastErrorMessage. That way, the error code of the disposed player private can be accessed after regeneration.
Let's see if this avoids the regressions.